ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.component.js

```diff
@@ -0,0 +1,84 @@
+import React, { useContext } from 'react';
+
+import PropTypes from 'prop-types';
+import { I18nContext } from '../../../contexts/i18n';
+
+import ToggleButton from '../../ui/toggle-button';
+import { Box, Text } from '../../component-library';
+import {
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+  FontWeight,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+
+import NetworkToggle from './NetworkToggle';
+import { useIncomingTransactionToggle } from './useIncomingTransactionToggle';
+
+const IncomingTransactionToggle = ({
+  wrapperRef,
+  incomingTransactionsPreferences,
+  allNetworks,
+  setIncomingTransactionsPreferences,
+}) => {
+  const t = useContext(I18nContext);
+
+  const {
+    networkPreferences,
+    isAllEnabled,
+    toggleAllEnabled,
+    toggleSingleNetwork,
+  } = useIncomingTransactionToggle({
+    incomingTransactionsPreferences,
+    allNetworks,
+    setIncomingTransactionsPreferences,
+  });
+
+  return (
+    <Box ref={wrapperRef}>
+      <Text variant={TextVariant.bodyLgMedium} fontWeight={FontWeight.Bold}>
+        {t('showIncomingTransactions')}
+      </Text>
+
+      <Text variant={TextVariant.bodyMd} color={TextColor.textAlternative}>
+        {t('showIncomingTransactionsInformation')}
+      </Text>
+      <Box
+        marginTop={3}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Row}
+        justifyContent={JustifyContent.spaceBetween}
+        data-testid="incoming-transaction-toggle-enable-all"
```

Comment:
If I turn the toggle ON for all the incoming transaction and turn any one toggle OFF then the toggle all button is off as well.
https://github.com/MetaMask/metamask-extension/assets/39872794/aa36335b-f2a2-436f-af11-80c50704db60

Reply:
Hey Nidhi, thanks for checking it out, and this is expected,
- any of sub toggle is off, toggleAll is off
- only when all of sub toggle is on, toggleAll is on